OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -130,7 +130,16 @@ static class DreamProcNativeRoot {
         public static DreamValue NativeProc_ckey(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
             string key = arguments.GetArgument(0, "Key").GetValueAsString();
 
-            key = Regex.Replace(key.ToLower(), "[^a-z]", ""); //Remove all punctuation and make lowercase
+            key = Regex.Replace(key.ToLower(), "[\\^]|[^a-z0-9@]", ""); //Remove all punctuation and make lowercase
+            return new DreamValue(key);
+        }
+
+        [DreamProc("ckeyEx")]
+        [DreamProcParameter("Key", Type = DreamValueType.String)]
+        public static DreamValue NativeProc_ckeyEx(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            string key = arguments.GetArgument(0, "Key").GetValueAsString();
+
+            key = Regex.Replace(key, "[\\^]|[^A-z0-9@_-]", ""); //Remove all punctuation except - and _
             return new DreamValue(key);
```

Comment:
According to the [reference page](http://www.byond.com/docs/ref/#/proc/ckeyEx), this parameter's name is `Text`
```suggestion
        [DreamProcParameter("Text", Type = DreamValueType.String)]
        public static DreamValue NativeProc_ckeyEx(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
            string text = arguments.GetArgument(0, "Text").GetValueAsString();

            text = Regex.Replace(text, "[\\^]|[^A-z0-9@_-]", ""); //Remove all punctuation except - and _
            return new DreamValue(text);
```